File: src/types/folder.ts
// Rust serializes struct fields as snake_case — TypeScript types match.

export interface Folder {
  id: number;
  name: string;
  parent_id: number | null;
  path: string;
  depth: number;
  created_at: string;
  updated_at: string;
}

/** Folder with its children populated — mirrors the Rust FolderNode response. */
export interface FolderNode extends Folder {
  children: FolderNode[];
}

File: src-tauri/src/models/folder.rs
use chrono::NaiveDateTime;

#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, sqlx::FromRow)]
pub struct Folder {
    pub id: i64,
    pub name: String,
    pub parent_id: Option<i64>,
    pub path: String,
    pub depth: i32,
    pub created_at: NaiveDateTime,
    pub updated_at: NaiveDateTime,
}

/// Folder with its children populated — used for the folder tree response.
#[derive(Debug, serde::Serialize)]
pub struct FolderNode {
    #[serde(flatten)]
    pub folder: Folder,
    pub children: Vec<FolderNode>,
}
